File: mobile/services/assistant.ts
import api from './api';

export const assistantService = {
  sendMessage: async (message: string) => {
    const response = await api.post('/assistant/message', { message });
    return response.data;
  },
  getChatHistory: async () => {
    const response = await api.get('/assistant/history');
    return response.data;
  },
  clearHistory: async () => {
    const response = await api.delete('/assistant/history');
    return response.data;
  }
};

File: backend/app/api/v1/assistant.py
from fastapi import APIRouter
from app.schemas.assistant import ChatRequest, ChatResponse
from app.schemas.common import APIResponse
from app.services.openai_service import OpenAIService

router = APIRouter()

@router.post("/chat", response_model=APIResponse)
async def chat(data: ChatRequest):
    response = await OpenAIService.chat(data.message, data.language)
    return APIResponse(data=ChatResponse(response=response, session_id=data.session_id or 1, language=data.language))

@router.get("/history", response_model=APIResponse)
async def history():
    return APIResponse(data=[])

@router.delete("/history", response_model=APIResponse)
async def delete_history():
    return APIResponse(message="Deleted successfully")
